tests/Integration/Behaviour/Features/Scenario/Product/specific_price_priorities.feature

```diff
@@ -0,0 +1,77 @@
+# ./vendor/bin/behat -c tests/Integration/Behaviour/behat.yml -s product --tags specific-price-priorities
+@reset-database-before-feature
+@specific-price-priorities
+Feature: Set Specific Price priorities from Back Office (BO).
+  As an employee I want to be able to set specific price priorities to single product and to all products
+
+  Scenario: I set specific price priorities to single product
+    Given I add product "product1" with following information:
+      | name       | en-US:pocket watch   |
+      | is_virtual | false                |
+    And product "product1" type should be standard
+    When I set following specific price priorities for product "product1":
+      | [id_country, id_currency, id_group, id_shop] |
+    Then product "product1" should have following specific price priorities:
+      | [id_country, id_currency, id_group, id_shop] |
+    When I set following specific price priorities for product "product1":
+      | [id_currency, id_country, id_group, id_shop] |
+    Then product "product1" should have following specific price priorities:
+      | [id_currency, id_country, id_group, id_shop] |
+
+  Scenario: I set specific price priorities to all products
+    Given I add product "product2" with following information:
+      | name       | en-US:golden wrist watch |
+      | is_virtual | false                    |
+    And I add product "product3" with following information:
+      | name       | en-US:silver wrist watch |
+      | is_virtual | false                    |
+    And product "product2" type should be standard
+    And product "product3" type should be standard
+    When I set following specific price priorities for all products:
+      | [id_country, id_currency, id_group, id_shop] |
+    Then product "product2" should have following specific price priorities:
+      | [id_country, id_currency, id_group, id_shop] |
+    And product "product3" should have following specific price priorities:
+      | [id_country, id_currency, id_group, id_shop] |
+
+  Scenario: Specific price priorities that was set to a single product
+    overrides the previously set global priorities (overrides priorities for this product only)
+    Given I set following specific price priorities for all products:
+      | [id_shop, id_currency, id_group, id_country] |
+    And product "product1" should have following specific price priorities:
```

Comment:
```suggestion
    Then product "product1" should have following specific price priorities:
```

Reply:
`And` just continues to state what was`Given`. I would only use `Then` after `When` to clearly state what is main action of the scenario.
I see it that way ,although I suppose it might vary depending on individual practice/taste :thinking: 
Anyway this line was modified to first assert all products priorities before taking any action, so it should not be relevant anymore

Reply:
Yep my bad I didn't realize it was part of the initial assertions

Reply:
Oh but I think the problem is that you use `Given` for an action statement (set the global priorities) But since it's an action `When` would be more appropriate (even if it's not the main part of your scenario)

I'd reserve `Given` for assertions like `Given this country is in database` or `Given the global properties are currently...` (to check the default value)

Reply:
I didn't read your last modifications before commenting here, indeed now that it's all initial assertions the use of `Given` makes more sense